caddyfile/parse_test.go

```diff
@@ -447,6 +447,24 @@ func TestParseAll(t *testing.T) {
 			{"localhost:1234"},
 		}},
 
+		{`localhost {
```

Comment:
I think we should move these tests into a seperate function making it obvious we are testing the final } issue.

Reply:
I agree